src/test/java/com/gradle/develocity/bamboo/ArtifactoryMaven3TaskMavenOptsSetterTest.java

```diff
@@ -46,7 +49,13 @@ void mavenOpsAreSet() {
         assertThat(configuration.get("key"), is(equalTo("value")));
         assertThat(
             configuration.get(ArtifactoryMaven3TaskMavenOptsSetter.MAVEN_OPTS_KEY),
-            is(equalTo("-Dmaven.ext.class.path=test/path/gradle-enterprise-maven-extension-1.15.4.jar -Dgradle.scan.uploadInBackground=false -Dgradle.enterprise.url=url"))
+            is(equalTo(
+                    "-Dmaven.ext.class.path=test/path/gradle-enterprise-maven-extension-1.15.4.jar " +
+                            "-Ddevelocity.scan.uploadInBackground=false " +
+                            "-Dgradle.scan.uploadInBackground=false " +
+                            "-Ddevelocity.url=url " +
+                            "-Dgradle.enterprise.url=url"
```

Comment:
So is it the case now that we will have both `-Ddevelocity.url` and `-Dgradle.enterprise.url` specified?

Reply:
Yes, that's the overall goal for Maven in all the CI plugins.

IIUC, we cannot always determine which Maven extension version is being used (@pshevche, please correct me, if I'm wrong with this). Therefore, our strategy is to configure the Maven extension in the CI plugins by specifying both the deprecated and new system properties.

In case the old (1.20 or before) Maven extension is being used, the new system properties `develocity.*` are simply ignored. In case the new (1.21 or after) Maven extension is being used, the new system properties win over the old ones. And we don't produce any deprecations warnings, as the old system properties are simply ignored.

Reply:
DV Maven Extension version is backed into this CI plugin. Once you update it to 1.21, that will be the version that will be injected. The compatibility matrix in README shows which extension versions are injected in which plugin versions. 
In other words, you don't need to worry about the old extension versions, as the plugin can only inject the one it's shipped with.

Reply:
Yes, but the embedded Maven extension is only applied, if we believe no other version of the Maven extension is present (see [here](https://github.com/gradle/develocity-bamboo-plugin/blob/snoopcheri/33446-configure-develocity-plugin/src/main/java/com/gradle/develocity/bamboo/MavenBuildScanInjector.java#L104-L107)).

But as far as I know, this logic is not fool-proof in all cases. That's why want to set both system property versions.

Reply:
You're right; some projects might have the extension already applied directly or indirectly, so it's better to make sure we're able to set the url we want